### ISSUE TITLE
Improve ARIA attributes

### ISSUE THREAD

**Issue by GauravKarakoti** [GauravKarakoti/Weather-API#153]
Some elements lack proper accessibility attributes (e.g., recent searches list).

**Comment by GauravKarakoti** [comment]
> Hello can you assign this issue to me

Go ahead
